### DISCUSSION TITLE
Failed to load credentials from the credentials provider

### DISCUSSION THREAD

**Discussion by pamaku** [awslabs/aws-sdk-rust#234] [Q&A] [answered]
Exploring posting  into s3 from ec2 instance within same VPC and wondering if SDK supports using IAM role instead of access-key and secrete-key. Any thoughts/guidance appreciated.

**Comment by rcoh** [comment, marked as answer]
Hello! Have you checked out the aws-config crate used in the examples? It
supports imds credentials

**Reply by pamaku** [reply]
Thanks will check that out

**Reply by pamaku** [reply]
Yes that did the trick.
